core/state/state_object.go

```diff
@@ -364,21 +315,30 @@ func (s *StateObject) updateTrie(db Database) Trie {
 			s.db.MetricsMux.Unlock()
 		}(time.Now())
 	}
+	// The snapshot storage map for the object
+	var (
+		storage map[common.Hash][]byte
+		origin  map[common.Hash][]byte
+		hasher  = crypto.NewKeccakState()
+	)
+	tr, err := s.getTrie()
+	if err != nil {
+		s.db.setError(err)
+		return nil, err
+	}
 	// Insert all the pending updates into the trie
-	tr := s.getTrie(db)
-
 	usedStorage := make([][]byte, 0, len(s.pendingStorage))
 	dirtyStorage := make(map[common.Hash][]byte)
 	for key, value := range s.pendingStorage {
 		// Skip noop changes, persist actual changes
 		if value == s.originStorage[key] {
 			continue
 		}
-		s.originStorage[key] = value
 		var v []byte
 		if value != (common.Hash{}) {
 			// Encoding []byte cannot fail, ok to ignore the error.
-			v, _ = rlp.EncodeToBytes(common.TrimLeftZeroes(value[:]))
+			value := value
```

Comment:
why "value := value" here?

Reply:
if no this line,  all `v` will be a slice of the last `value` in s.pendingStorage.
the difference is that:
as before, EncodeToBytes will do a copy
now, no `EncodeToBytes` will be called, so `value := value` do the copy

Reply:
if no this line,  all `v` will be a slice of the last `value` in s.pendingStorage.
the difference is that:
as before, EncodeToBytes will do a copy
now, no `EncodeToBytes` will be called, so `value := value` do the copy